include/nbl/builtin/hlsl/emulated/emulated_float64_t_impl.hlsl

```diff
@@ -0,0 +1,327 @@
+#ifndef _NBL_BUILTIN_HLSL_EMULATED_FLOAT64_T_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_EMULATED_FLOAT64_T_IMPL_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/ieee754/ieee754.hlsl>
+#include <nbl/builtin/hlsl/algorithm.hlsl>
+#include <nbl/builtin/hlsl/tgmath.hlsl>
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+
+// TODO: when it will be possible, use this unions wherever they fit:
+/*
+* union Mantissa
+* {
+*   struct
+*   {
+*       uint32_t highBits;
+*       uint64_t lowBits;
+*   };
+*
+*   uint32_t2 packed;
+* };
+*
+*/
+/*
+* union Mantissa
+* {
+*   struct
+*   {
+*       uint64_t lhs;
+*       uint64_t rhs;
+*   };
+*
+*   uint32_t4 packed;
+* };
+*
+*/
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace impl
```

Comment:
you might want to make the namespace `emulated_float64_impl` (or make a `struct emulated_float64_base` and turn everything into statics) because with the function names you have, we can have a lot of collisions eventually in the `nbl::hlsl:impl` namespace